lib/Test/MockFile.pm

```diff
@@ -1712,6 +1714,120 @@ BEGIN {
         $mock->{'has_content'} = undef;
         return 1;
     };
+
+    *CORE::GLOBAL::chown = sub(@) {
+        my ( $uid, $gid, @files ) = @_;
+
+        $^O eq 'MSWin32'
+            and return 0; # does nothing on Windows
+
+        # Not an error, report we changed zero files
+        @files
+            or return 0;
+
+        my %mocked_files = map +( $_ => _get_file_object($_) ), @files;
+
+        # The idea is that if some are mocked and some are not,
+        # it's probably a mistake
+        my $mocked_count = grep ref, values %mocked_files;
+        if ( $mocked_count && $mocked_count != @files ) {
+            die "You called chown() on a mix of mocked and unmocked files - this is very likely a bug on your side";
```

Comment:
maybe list the files?

Reply:
or provide some context to know where it's coming from?
as most of the time it s going to be action at a distance

Reply:
Good idea. Will do.